std/experimental/logger/core.d

```diff
@@ -750,14 +748,42 @@ abstract class Logger
         string moduleName;
         /// the $(D LogLevel) associated with the log message
         LogLevel logLevel;
-        /// thread id of the log message
-        Tid threadId;
         /// the time the message was logged
         SysTime timestamp;
         /// the message of the log message
         string msg;
-        /// A refernce to the $(D Logger) used to create this $(D LogEntry)
+    }
+
+    /** LogEntry is an aggregation combining all information associated
+    with a log message. This aggregation will be passed to the method
+    writeLogMsg.
+    */
+    protected struct LogEntry
+    {
+        LogEntryBase base;
+        alias base this;
+
+        /// thread id of the log message
+        Tid threadId;
+        /// A reference to the $(D Logger) used to create this $(D LogEntry)
         Logger logger;
+
+        this(string file = string.init, int line = int.init,
+            string funcName = string.init, string prettyFuncName = string.init,
+            string moduleName = string.init, LogLevel logLevel = LogLevel.init,
+            Tid threadId = Tid.init, SysTime timestamp = SysTime.init,
+            string msg = string.init, Logger logger = Logger.init) @safe
```

Comment:
As-is, this constructor isn't even documented, so I would strongly recommend not commiting to such a complicated interface (this is akin to adding a constructor with 10 overloads). Just make the parameters required.